--- a/vicinae/src/extension/extension-grid-model.hpp
+++ b/vicinae/src/extension/extension-grid-model.hpp
@@ -94,7 +94,7 @@
 
       std::initializer_list<WS> fields = {WS{item.title, TITLE_WEIGHT}, WS{item.subtitle, SUBTITLE_WEIGHT}};
       auto kws = item.keywords | std::views::transform([](auto &&s) { return WS{s, KEYWORD_WEIGHT}; });
-      auto ss = std::views::concat(fields, kws);
+      auto ss = std::views::join(std::array{fields, kws});
       int score = fzf::defaultMatcher.fuzzy_match_v2_score_query(fields, q, false);
 
       return ScoredItem{.item = &item, .score = score};
