--- a/vicinae/src/services/root-item-manager/root-item-manager.cpp
+++ b/vicinae/src/services/root-item-manager/root-item-manager.cpp
@@ -149,7 +149,7 @@
   std::string alias = meta->alias.value_or("");
   std::initializer_list<WS> ss = {{title, 1.0f}, {subtitle, 0.6f}, {alias, 1.0f}};
   auto kws = keywords | std::views::transform([](auto &&kw) { return WS{kw, 0.3f}; });
-  auto strs = std::views::concat(ss, kws);
+  auto strs = std::views::join(std::array{ss, kws});
   float score = pattern.empty() ? 1 : fzf::defaultMatcher.fuzzy_match_v2_score_query(strs, pattern, false);
   double frequencyScore = std::log(1 + meta->visitCount * 0.1);
   float frequencyWeight = 0.2;
@@ -416,7 +416,7 @@
 
   if (!provider || !item) return {};
 
-  return std::views::concat(provider->preferences(), item->preferences()) | std::ranges::to<std::vector>();
+  return std::views::join(std::array{provider->preferences(), item->preferences()}) | std::ranges::to<std::vector>();
 }
 
 QJsonObject RootItemManager::getPreferenceValues(const EntrypointId &id) const {
