--- a/crates/settings_ui/src/page_data.rs
+++ b/crates/settings_ui/src/page_data.rs
@@ -10,10 +10,10 @@
     pages::render_edit_prediction_setup_page,
 };
 
-const DEFAULT_STRING: String = String::new();
+static DEFAULT_STRING: String = String::new();
 /// A default empty string reference. Useful in `pick` functions for cases either in dynamic item fields, or when dealing with `settings::Maybe`
 /// to avoid the "NO DEFAULT" case.
-const DEFAULT_EMPTY_STRING: Option<&String> = Some(&DEFAULT_STRING);
+static DEFAULT_EMPTY_STRING: Option<&String> = Some(&DEFAULT_STRING);
 
 macro_rules! concat_sections {
     (@vec, $($arr:expr),+ $(,)?) => {{
