diff --git a/sdk/runtime/BUILD.gn b/sdk/runtime/BUILD.gn
index 8cd05ff445c..96a69817af1 100644
--- a/sdk/runtime/BUILD.gn
+++ b/sdk/runtime/BUILD.gn
@@ -297,7 +297,6 @@ config("dart_config") {
   } else {
     cflags = [
       "-Wno-unused-parameter",
-      "-Wno-unused-private-field",
       "-Wnon-virtual-dtor",
       "-Wvla",
       "-Woverloaded-virtual",
@@ -313,11 +312,13 @@ config("dart_config") {
         "-Wimplicit-fallthrough",
         "-fno-strict-vtable-pointers",  # Handle assignment updates vtable
                                         # pointers.
+        "-Wno-unused-private-field",
       ]
     } else {
       cflags += [
         "-Wno-cast-function-type",
         "-Wno-dangling-reference",
+        "-Wno-unused-but-set-variable",
       ]
     }
 
diff --git a/sdk/third_party/binaryen/BUILD.gn b/sdk/third_party/binaryen/BUILD.gn
index ebdfa32ccb3..d95d2b9c92e 100644
--- a/sdk/third_party/binaryen/BUILD.gn
+++ b/sdk/third_party/binaryen/BUILD.gn
@@ -31,6 +31,7 @@ config("binaryen_flags") {
       # GCC
       cflags += [
         "-Wno-redundant-move",
+        "-Wno-unused-but-set-variable",
         "-Wno-uninitialized",
         "-Wno-init-list-lifetime",
 
@@ -38,6 +39,7 @@ config("binaryen_flags") {
         # https://github.com/WebAssembly/binaryen/blob/cdb7aeab40b4c522de20b242019f7e88641445d5/src/wasm/wasm-type.cpp#L530.
         "-Wno-maybe-uninitialized",
         "-Wno-deprecated",
+        "-Wno-array-bounds",
       ]
     }
   }
